Sources/TuistCache/ContentHashing/CacheProfileContentHasher.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+import TuistCore
+import TuistGraph
+
+public protocol CacheProfileContentHashing {
+    func hash(cacheProfile: TuistGraph.Cache.Profile) throws -> String
+}
+
+/// `CacheProfileContentHasher`
+/// is responsible for computing a unique hash that identifies a caching profile
+public final class CacheProfileContentHasher: CacheProfileContentHashing {
+    private let contentHasher: ContentHashing
+
+    // MARK: - Init
+
+    public init(contentHasher: ContentHashing) {
+        self.contentHasher = contentHasher
+    }
+
+    // MARK: - CacheProfileContentHashing
+
+    public func hash(cacheProfile: TuistGraph.Cache.Profile) throws -> String {
+        let stringsToHash = [cacheProfile.name, cacheProfile.configuration]
```

Comment:
might be worth writing to the internal documentation of `Cache.Profile` a warning that if one adds a property, it should be added here, too. I also thought about how we could check this statically but only thing that could probably achieve this is using `Mirror`, so I think some written warning should do

Reply:
Sure, good idea, I added a warning message to the `Cache.Profile` struct 👍